### PR TITLE
Add banner and accent color to user and some fixes/improvements

### LINKED ISSUE #74: Support user banner and accent color
# Summary
The user object should support the banner and accent color fields.

Docs can be found [here](https://github.com/discord/discord-api-docs/pull/3448)

### PR DESCRIPTION
**Pull request by INikonI**
+ Resolve #74 
+ Mini refactor of color class and add to them implicit operators for `uint` and const with max decimal value
+ Some fixes on `Test.Unit`

### CODE REVIEW

File: src/Discord.Net.Core/Entities/Users/IUser.cs [hunk 1]
```diff
@@ -34,6 +46,16 @@ public interface IUser : ISnowflakeEntity, IMentionable, IPresence
...
         /// <summary>
+        ///     Gets the banner URL for this user.
+        /// </summary>
+        /// <param name="format">The format to return.</param>
+        /// <param name="size">The size of the image to return in. This can be any power of two between 16 and 2048.
+        /// </param>
+        /// <returns>
+        ///     A string representing the user's avatar URL; <c>null</c> if the user does not have an banner in place.
+        /// </returns>
+        string GetBannerUrl(ImageFormat format = ImageFormat.Auto, ushort size = 128);
```
Comment (quinchs) [inline src/Discord.Net.Core/Entities/Users/IUser.cs]: since banners are bigger images, shouldn't the default size be bigger?

also is the size of the image the width or height?

File: src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs [hunk 2]
```diff
@@ -53,6 +53,20 @@ public override string AvatarId
             internal set => GuildUser.AvatarId = value;
         }
 
+        /// <inheritdoc/>
+        public override string BannerId
+        {
+            get => GuildUser.BannerId;
+            internal set => GuildUser.BannerId = value;
```
Comment (quinchs) [inline src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs:60]: why add setters if they are not used?

File: src/Discord.Net.WebSocket/Entities/Users/SocketWebhookUser.cs [hunk 3]
```diff
@@ -24,6 +24,13 @@ public class SocketWebhookUser : SocketUser, IWebhookUser
         public override ushort DiscriminatorValue { get; internal set; }
         /// <inheritdoc />
         public override string AvatarId { get; internal set; }
+
+        /// <inheritdoc />
+        public override string BannerId { get; internal set; }
+
+        /// <inheritdoc />
+        public override Color? AccentColor { get; internal set; }
```
Comment (quinchs) [inline src/Discord.Net.WebSocket/Entities/Users/SocketWebhookUser.cs]: webhooks cannot have banner/accent colors, please make it throw a NotSupportedException

**Review by quinchs — CHANGES_REQUESTED** [review]

File: src/Discord.Net.WebSocket/Entities/Users/SocketWebhookUser.cs (hunk 3 above)
Comment (INikonI) [reply src/Discord.Net.WebSocket/Entities/Users/SocketWebhookUser.cs]: its just `null` for bots and webhooks? 🤔

File: src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs (hunk 2 above)
Comment (INikonI) [reply src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs:60]: idk, `AvatarId` property too has setter

File: src/Discord.Net.Core/Entities/Users/IUser.cs (hunk 1 above)
Comment (INikonI) [reply src/Discord.Net.Core/Entities/Users/IUser.cs]: > since banners are bigger images, shouldn't the default size be bigger?

increased in b8b92b7be8470a3aad8d7bc25f41028be9ce2d12, but 128 is default width for images in discord

File: src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs (hunk 2 above)
Comment (INikonI) [reply src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs:60]: Because it implementation of interfaces/classes, if specifically then implementation of `SocketUser`.

File: src/Discord.Net.WebSocket/Entities/Users/SocketWebhookUser.cs (hunk 3 above)
Comment (INikonI) [reply src/Discord.Net.WebSocket/Entities/Users/SocketWebhookUser.cs]: Resolved in abcb4e7a9a524d79d54ff980629ffebb36ae54f7

File: src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs (hunk 2 above)
Comment (quinchs) [reply src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs:60]: Any property like that will be updated in the class that owns it, therfor we dont need to modify any properties out of sync with that guild user. Could you please remove it

Comment (INikonI) [reply src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs:60]: ok

Comment (INikonI) [reply src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs:60]: but how 🤔

Comment (INikonI) [reply src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs:60]: This class created by you, and inherits `SocketUser`. `AvatarID` and other props also has setters, because its implementation of `SocketUser`.

Comment (quinchs) [reply src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs:60]: just dont override it...

Comment (quinchs) [reply src/Discord.Net.WebSocket/Entities/Users/SocketThreadUser.cs:60]: nvm its good ig.

File: src/Discord.Net.Core/Entities/Roles/Color.cs [hunk 4]
```diff
@@ -10,68 +10,70 @@ namespace Discord
...
         /// <returns> A color struct with the hex value of <see href="http://www.color-hex.com/color/71368A">71368A</see>.</returns>
-        public static readonly Color DarkPurple = new Color(0x71368A);
+        public static readonly Color DarkPurple = new(0x71368A);
         /// <summary> Gets the magenta color value. </summary>
         /// <returns> A color struct with the hex value of <see href="http://www.color-hex.com/color/E91E63">E91E63</see>.</returns>
-        public static readonly Color Magenta = new Color(0xE91E63);
+        public static readonly Color Magenta = new(0xE91E63);
         /// <summary> Gets the dark magenta color value. </summary>
         /// <returns> A color struct with the hex value of <see href="http://www.color-hex.com/color/AD1457">AD1457</see>.</returns>
-        public static readonly Color DarkMagenta = new Color(0xAD1457);
```
Comment (quinchs) [inline src/Discord.Net.Core/Entities/Roles/Color.cs:44]: Why did you change the constructor?

**Review by quinchs — CHANGES_REQUESTED** [review]

File: src/Discord.Net.Core/Entities/Roles/Color.cs (hunk 4 above)
Comment (INikonI) [reply src/Discord.Net.Core/Entities/Roles/Color.cs:44]: Redundant type specification

**Comment by quinchs** [comment]
Please fix the conflicts in `test/Discord.Net.Tests.Unit/GuildPermissionsTests.cs`

**Comment by INikonI** [comment]
Reason of 2370f62c6fc0d99e22aeae5a35ae06775212bdf9 commit 
image
